### PR TITLE
add RunRepeatedly/DecisionOptionName  to WorkflowEventEntity

### PR DESCRIPTION
**Pull request by MehdyKarimpour**

### CODE REVIEW

**Comment by MehdyKarimpour** [comment]
Hi Olmo,

First requirement is the need to send Email/SMS or do something in parallel while a case activity is in pending state. 
First idea is using Non-interrupting(Fork) boundary timers, but actually they run only once but we need to run them repeatedly while the case activity is pending. So I have added a new property `bool RunRepeatedly` to WorkflowEventEntity with default false for full backward compatibility and consider it on CaseActivityTask.Timeout and CaseActivityOperation.Timer. We need to use LastExecution when RunRepeatedly is true and the rest changes. Also adapt the UI that it makes sense only for Non-interrupting(Fork) timers.

Second requirement is to simulate doing Accept/Decline/... (or other configurable Decision Task Options) automatically. 
Now it is possible to do it by a Interrupting-timer for Tasks (Next) but not for DecisionTasks (Accept/Decline/...). Also it is not possible to do multiple Accept/Decline/... from Inbox. The idea is to use Interrupting-timer for it. So, I have added another new property called `string? DecisionOptionName` to WorkflowEventEntity and consider it on CaseActivityLogic.ExecuteBoundaryTimer(). 
Because it is rare to use, I didn`t use <select /> for the new property, but I have added some validations for it. 

Hopefully this comments will useful, thanks.

File: Signum.Engine.Extensions/Workflow/WorkflowNodeGraph.cs [hunk 1]
```diff
@@ -238,6 +238,19 @@ public void Validate(List<WorkflowIssue> issuesContainer, Action<WorkflowGateway
 
                 if (e.Type == WorkflowEventType.IntermediateTimer && !e.Name.HasText())
                     issues.AddError(e, WorkflowValidationMessage.IntermediateTimer0ShouldHaveName.NiceToString(e));
+
+                if (e.Type == WorkflowEventType.BoundaryInterruptingTimer)
+                {
+                    var parentActivity = Activities.Values.Where(a => a.BoundaryTimers.Contains(e)).SingleEx();
+                    if (string.IsNullOrWhiteSpace(e.DecisionOptionName) && parentActivity.Type == WorkflowActivityType.Decision)
```
Comment (olmobrutall) [inline Signum.Engine.Extensions/Workflow/WorkflowNodeGraph.cs:245]: The new field becomes now mandatory for InterruptingTimers on Decision activities. Right?

File: Signum.Entities.Extensions/Workflow/WorkflowEvent.cs [hunk 2]
```diff
@@ -47,11 +54,24 @@ public void SetModel(ModelEntity model)
...
         this.Timer = wModel.Timer;
         this.BpmnElementId = wModel.BpmnElementId;
         this.CopyMixinsFrom(wModel);
         //WorkflowEventTaskModel.ApplyModel(this, wModel.Task);
     }
+
+    protected override string? PropertyValidation(PropertyInfo pi)
+    {
+        if (pi.Name == nameof(RunRepeatedly) && RunRepeatedly && Type != WorkflowEventType.BoundaryForkTimer)
+            RunRepeatedly = false;
```
Comment (olmobrutall) [inline Signum.Entities.Extensions/Workflow/WorkflowEvent.cs:68]: Changing fields in Property Validation is not so cool. Maybe move it to PreSaving?

File: Signum.Engine.Extensions/Workflow/WorkflowNodeGraph.cs (hunk 1 above)
Comment (MehdyKarimpour) [reply Signum.Engine.Extensions/Workflow/WorkflowNodeGraph.cs:245]: yes

File: Signum.Entities.Extensions/Workflow/WorkflowEvent.cs (hunk 2 above)
Comment (MehdyKarimpour) [reply Signum.Entities.Extensions/Workflow/WorkflowEvent.cs:69]: done and pushed

**Comment by olmobrutall** [comment]
Thanks for the changes!

**Comment by MehdyKarimpour** [comment]
your welcome 👍

**Pull request merged**